src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java

```diff
@@ -2799,6 +2815,31 @@ public Problem loadCCSProblemFiles(IInternalContest contest, String dataFileBase
         return problem;
     }
 
+    protected void loadDataFiles(Problem problem, ArrayList<SerializedFile> dataFiles, ArrayList<SerializedFile> answerFiles, String sampleDataDirectory, boolean loadExternalFile) {
+        
+        String[] inputFileNames = getFileNames(sampleDataDirectory, ".in");
+        String[] answerFileNames = getFileNames(sampleDataDirectory, ".ans");
+
+        Arrays.sort(inputFileNames);
+        
+        for (int idx = 0; idx < inputFileNames.length; idx++) {
+            
+            problem.addTestCaseFilenames(inputFileNames[idx], answerFileNames[idx]);
```

Comment:
Hmm... this line of code references element `[idx]` in both the `inputFileNames` and `answerFilesNames` arrays.  The first one (`inputFileNames`) has been sorted (line 2823, above), but the second one has not.  Doesn't that mean that the elements at `[idx]` in the two arrays are no longer guaranteed to be corresponding?

Reply:
Although I see that this is also the way it was done when only the "secret" files were being loaded, so (presumably?) it works... although I'm not sure how.

Reply:
Yes, I followed the logic through, and unless I missed something, it would only work if the directory is already in sorted order.  There is also code in that "idx" loop that creates 2 arrays (dataFiles and answerFiles) for use just below in computing "problemDataFiles", which is passed into loadCCSProblemFiles.  It turns out that problemDataFiles is never used by the caller.  Evidently it WAS used at one point for the ClicsOutputValidator, but it is no longer used in there, so that code is dunsel.  Note, however, that the code that calculates dataFiles and answerFiles does go out of its way to compute the CORRECT answer file name based on the input file name. `String answerShortFileName = inputFileNames[idx].replaceAll(".in$", ".ans");`  However, as I said, it is for naught, since those arrays serve no useful purpose.  Similar code should probably be used to form the answer file name before the call to `problem.addTestCaseFilenames()`.  Perhaps:

`problem.addTestCaseFilenames(inputFileNames[idx], inputFileNames[idx].replaceAll(".in", ".ans"));`

Reply:
@johnbrvc  The original issue which I pointed out -- the fact that `inputFileNames` was being sorted while `answerFileNames` was not -- has been fixed (by the addition of a call to `Arrays.sort(answerFileNames)`.  However, I'm not sure whether to mark this conversation "resolved" because it appears maybe you introduced an addition issue in your comment.  What's your take -- does the addition of the call to `Arrays.sort()` fix the problem, or is there still another problem (besides "dunsel code") ?